test/unit/math/opencl/rev/multiply_test.cpp

```diff
@@ -0,0 +1,32 @@
+#ifdef STAN_OPENCL
+#include <stan/math/opencl/rev/opencl.hpp>
+#include <stan/math.hpp>
+#include <gtest/gtest.h>
+#include <test/unit/math/opencl/util.hpp>
+
+auto matrix_multiply_functor
+    = [](const auto& a, const auto& b) { return a * b; };
+
+TEST(OpenCLMatrixMultiply, prim_rev_values_small) {
+  int N = 2;
+  int M = 3;
+  int K = 4;
+
+  Eigen::MatrixXd a(N, M);
+  a << 1, 2, 3, 4, 5, 6;
+  Eigen::MatrixXd b(M, K);
+  b << 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1;
+  stan::math::test::compare_cpu_opencl_prim_rev(matrix_multiply_functor, a, b);
```

Comment:
As far as I understand, the tests only tests multiply(matrix_cl, matrix_cl) and multply(var_value, var_value). We need to also test multply(matrix_cl, var_value) multply(var_value, matrix_cl).